### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Katerina-codes**
- Improve naming.
- Handle game in progress.
- Game updates the board & gets move from GUI player. Controller class makes GUI player use game's run() method for all behaviour.
- Implement game mode for unbeatable computer player goes first.
- Add tests to test the Game and Gui Classes behave as expected with the Controller class.

@mollies @jsuchy

### CODE REVIEW

File: src/main/java/CommandLineUI.java [hunk 1]
```diff
@@ -98,6 +98,11 @@ public void displayBoard(List<Mark> grid, int size) {
         output.println(formattedRows);
     }
 
+    @Override
+    public void announceGameStatus(Result winner) {
+
+    }
+
```
Comment (MollieS) [inline src/main/java/CommandLineUI.java]: why does this have to be here?

File: src/main/java/game/Controller.java [hunk 2]
```diff
@@ -22,70 +24,46 @@
...
+    public Button computerVsGuiPlayer;
     private Board board = new Board(3);
-    private final Game game = new Game(this, board);
     private GuiPlayer guiPlayer = new GuiPlayer(X);
 
-    @Override
-    public void askForGameMode() {
-        askForGameMode.setText("Press 8 for Human Vs Human\nEnter 9 for Human Vs Computer");
+    public Controller(Game game) {
+        this.game = game;
```
Comment (MollieS) [inline src/main/java/game/Controller.java]: I think this can go now

File: src/main/java/game/Players/Player.java [hunk 3]
```diff
@@ -9,4 +9,6 @@
     Mark getMark();
 
     boolean hasMove();
+
+//    void receiveMove(int moveNumber);
```
Comment (MollieS) [inline src/main/java/game/Players/Player.java]: 👎

File: src/main/java/CommandLineUI.java [hunk 4]
```diff
@@ -98,6 +98,11 @@ public void displayBoard(List<Mark> grid, int size) {
         output.println(formattedRows);
     }
 
+    @Override
+    public void announceGameStatus(Result winner) {
+
```
Comment (fabientownsend) [inline src/main/java/CommandLineUI.java]: Do you need to Override despite you don't implement anything?

Comment (Katerina-codes) [reply src/main/java/CommandLineUI.java]: Good point, thank you Fabien :)

**Pull request merged**